src/DIRAC/Interfaces/API/Dirac.py

```diff
@@ -139,41 +147,72 @@ def submitJob(self, job, mode="wms"):
         """
         self.__printInfo()
 
+        if not isinstance(job, (str, Job)):
+            return S_ERROR("The job must be either a string or a job object")
+
         if isinstance(job, str):
             if os.path.exists(job):
                 self.log.verbose(f"Found job JDL file {job}")
                 with open(job) as fd:
-                    jdlAsString = fd.read()
+                    jdl = fd.read()
             else:
                 self.log.verbose("Job is a JDL string")
-                jdlAsString = job
-            jobDescriptionObject = None
-        else:  # we assume it is of type "DIRAC.Interfaces.API.Job.Job"
-            try:
-                formulationErrors = job.errorDict
-            except AttributeError as x:
-                self.log.verbose(f"Could not obtain job errors:{x}")
-                formulationErrors = {}
-
-            if formulationErrors:
-                for method, errorList in formulationErrors.items():
-                    self.log.error(">>>> Error in {}() <<<<\n{}".format(method, "\n".join(errorList)))
-                return S_ERROR(formulationErrors)
-
-            # Run any VO specific checks if desired prior to submission, this may or may not be overridden
-            # in a derived class for example
-            try:
-                result = self.preSubmissionChecks(job, mode)
-                if not result["OK"]:
-                    self.log.error(f"Pre-submission checks failed for job with message: \"{result['Message']}\"")
-                    return result
-            except Exception as x:
-                msg = f'Error in VO specific function preSubmissionChecks: "{x}"'
-                self.log.error(msg)
-                return S_ERROR(msg)
+                jdl = job
+
+            jdl = jdl.strip()
+            if not jdl.startswith("[") and not jdl.endswith("]"):
+                jdl = f"[{jdl}]"
+
+            jobDescription = ClassAd(jdl)
+
+            jdl = transformParametricJobIntoParsableOne(jobDescription)
+
+            res = checkIfParametricJobIsCorrect(jobDescription)
+            if not res["OK"]:
+                return res
+
+            stdout = "std.out"
+            if jobDescription.lookupAttribute("StdOutput"):
+                stdout = jobDescription.getAttributeString("StdOutput")
+                jobDescription.deleteAttribute("StdOutput")
+
+            stderr = "std.err"
+            if jobDescription.lookupAttribute("StdError"):
+                stderr = jobDescription.getAttributeString("StdError")
+                jobDescription.deleteAttribute("StdError")
+
+            job = Job(stdout=stdout, stderr=stderr)
+
+            res = loadJDLasJob(job, jobDescription)
+            if not res["OK"]:
+                return res
+            job = res["Value"]
+
+        try:
+            formulationErrors = job.errorDict
+        except AttributeError as x:
+            self.log.verbose(f"Found job JDL file: {job}")
```

Comment:
Not sure to understand this log.
A suggestion:
```suggestion
            self.log.verbose(f"Could not obtain job errors: {x}")
```